### PR TITLE
Devices json update rebased

### PR DESCRIPTION
**Pull request by julianstirling**
@jmwright This is a rebase of your [branch](https://github.com/Wakoma/nimble/pull/52) but rebased on top of the new changes in master. I pushed to a new branch as I thought it rude to force-push over your branch in case I break some work in progress.

On rebasing I

* Moved your updater script into the new orchestration module.
* Deleted `example-run.py` as it is based on the old orchestration. *I know this killed your hashing code, see [issue on hashing](https://github.com/Wakoma/nimble/issues/54).*
* Updated `generate.py` so it works with the new `devices.json`

Are you happy for this to be merged? Or do you want to rebase differently, or carry on your draft in another way?

### CODE REVIEW

**Comment by julianstirling** [comment]
FYI I added the requirement for the changes to be reviewed and approved.

File: nimble_orchestration/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+The files in this directory may not be directly imported within the codebase, but are needed anyway. Some may be applications that are run manually by a user, for instance `devices_json_updater.py`.
+
+### devices_json_updater.py
+
+This converts the CSV that is downloaded from NocoDB into something that can be used by the orchastration/generation scripts. The scripts use the `devices.json` file that is in the root of this repository. This should eventually be replaced by a direct connection via the NocoDB API. This utility requires that it be passed the path to the CSV file to convert.
```
Comment (jmwright) [inline nimble_orchestration/README.md:5]:
```suggestion
This converts the CSV that is downloaded from NocoDB into something that can be used by the orchestration/generation scripts. The scripts use the `devices.json` file that is in the root of this repository. This should eventually be replaced by a direct connection via the NocoDB API. This utility requires that it be passed the path to the CSV file to convert.
```

**Comment by jmwright** [comment]
@julianstirling Please also remove `devices.json.old` since I was just keeping that in case I needed to revert during the update. I also made an inline suggestion that corrects a spelling mistake that I made in the markdown. Other than that it should be ready to merge.

**Comment by julianstirling** [comment]
Thanks @jmwright. I think with the new review rules you need to select the approve option in the review tab

**Review by jmwright — APPROVED** [review]
Thanks @julianstirling

**Pull request merged**